### ISSUE TITLE
opbeans-ruby transactions have spurious parent ID

### ISSUE THREAD

**Issue by axw** [elastic/apm-agent-ruby#1231]
Transactions reported by opbeans-ruby all have a non-empty `parent.id` which does not correspond to any span or transaction. This results in traces without any root transaction, and breaks tail-based sampling which operates on root transactions.

Steps to reproduce:

1. In apm-integration-testing, run `./scripts/compose.py start 8.0.0-rc2 --bc --with-opbeans-ruby`
2. Open the APM UI, navigate to the `opbeans-ruby` service
3. Select the only transaction group, `ApplicationController#index`
4. Enter `not parent.id:*` in the search bar
5. Observe that there are no trace samples

**Comment by felixbarny** [comment]
May be related to the usage of `ensure_parent_id`.

https://github.com/elastic/opbeans-ruby/blob/5c2226d5443325e80171fff881c2a54f294746ee/app/controllers/application_controller.rb#L41

> Enter not parent.id:* in the search bar
> Observe that there are no trace samples

Isn't that expected as the parent would come from RUM?

**Comment by axw** [comment]
Good point. I didn't have RUM configured, so that explains why there were no root transactions. We should probably only call `ensure_parent_is` when the frontend is RUM-instrumented.

That raises the question of why there were only `ApplicationController#index` transactions though. Shouldn't there be other ones, e.g. for `/api/orders`? And those ones ones shouldn't invoke `ensure_parent_id`.

**Comment by felixbarny** [comment]
I haven't tried with apm-integration-testing but on edge, there were no `ApplicationController#index` transactions.

**Comment by felixbarny** [comment]
Right, I can see that too now. Seems like the RUM integration setup doesn't work as expected.

**Comment by smith** [comment]
Closing this issue on the agent repo. Haven't checked opbeans-ruby but that can be fixed independently if needed.
